staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -682,7 +693,7 @@ func (qs *queueSet) selectQueueLocked() *queue {
 	// queue here. if the last virtual start time (excluded estimated cost)
 	// falls behind the global virtual time, we update the latest virtual
 	// start by: <latest global virtual time> + <previously estimated cost>
-	previouslyEstimatedServiceTime := float64(minQueue.requestsExecuting) * qs.estimatedServiceTime
+	previouslyEstimatedServiceTime := float64(minQueue.seatsInUse) * qs.estimatedServiceTime
```

Comment:
selectQueueLocked currently doesn't do what is decribed in KEP.
With the current code it would dispatch the request even if we don't have enough available seats.

We should change that so that:
- it is chosing the queue from which the request should be dispatched
- if it doesn't have enough seats, returns nil (but remembers that a queue was chosen already)
- in the next call, if the queue was chosen previously but not dispatched), tries only that queue

Reply:
I assume this comment was made because the KEP does not describe the current code.  https://github.com/kubernetes/enhancements/pull/2766 restores the description of the current code, among other things.